File: solutions/cpp/0088-merge-sorted-array.cpp
class Solution {
public:
    void merge(vector<int>& nums1, int m, vector<int>& nums2, int n) {
        int p1 = m - 1;
        int p2 = n - 1;
        int p = m + n - 1;
        
        while (p1 >= 0 && p2 >= 0) {
            if (nums1[p1] > nums2[p2]) {
                nums1[p] = nums1[p1];
                p1--;
            } else {
                nums1[p] = nums2[p2];
                p2--;
            }
            p--;
        }
        
        // add remaining elements from nums2
        for (int i = 0; i <= p2; i++) {
            nums1[i] = nums2[i];
        }
    }
};

File: solutions/rust/0088-merge-sorted-array.rs
impl Solution {
    pub fn merge(nums1: &mut Vec<i32>, m: i32, nums2: &mut Vec<i32>, n: i32) {
        let (mut p1, mut p2) = (m as usize, n as usize);
        
        while p2 > 0 {
            if p1 > 0 && nums1[p1 - 1] > nums2[p2 - 1] {
                nums1[p1 + p2 - 1] = nums1[p1 - 1];
                p1 -= 1;
            } else {
                nums1[p1 + p2 - 1] = nums2[p2 - 1];
                p2 -= 1;
            }
        }
    }
}
